mmpretrain/datasets/caltech101.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import List
+
+from mmengine import get_file_backend, list_from_file
+
+from mmpretrain.registry import DATASETS
+from .base_dataset import BaseDataset
+from .categories import CALTECH101_CATEGORIES
+
+
+@DATASETS.register_module()
+class Caltech101(BaseDataset):
+    """The Caltech101 Dataset.
+
+    Support the `Caltech101 <https://data.caltech.edu/records/mzrjq-6wc02>`_ Dataset.
+    After downloading and decompression, the dataset directory structure is as follows.
+
+    Caltech101 dataset directory: ::
+
+        Caltech101 (data_root)/
+        ├── 101_ObjectCategories
+        │   ├── class_x
+        │   │   ├── xx1.jpg
+        │   │   ├── xx2.jpg
+        │   │   └── ...
+        │   ├── class_y
+        │   │   ├── yy1.jpg
+        │   │   ├── yy2.jpg
+        │   │   └── ...
+        │   └── ...
+        ├── meta
+        │   ├── train.txt
+        │   └── test.txt
+        └── ....
+
+    Please note that since there is no official splitting for training and
+    test set, you can use the train.txt and text.txt provided by us or
+    create your own annotation files.
+
+    The download link for annotations: <https://xxxx>.
+
+    Args:
+        data_root (str): The root directory for the Caltech101 dataset.
+        split (str, optional): The dataset split, supports "train" and "test".
+            Default to "train".
+
+    Examples:
+        >>> from mmpretrain.datasets import Caltech101
+        >>> train_cfg = dict(data_root='data/Caltech', split='train')
+        >>> train = Caltech101(**train_cfg)
+        >>> train
+        Dataset Caltech101
+            Number of samples:  3060
+            Number of categories:       102
+            Root of dataset:    data/Caltech
+        >>> test_cfg = dict(data_root='data/Caltech', split='test')
+        >>> test = Caltech101(**test_cfg)
```

Comment:
```test_dataset = Caltech101(data_root='data/Caltech', split='test')``` is enough. The current configuration items are relatively short, the lines about 'test_cfg', 'train_cfg' can be removed.

```
        >>> from mmpretrain.datasets import Caltech101
        >>> Caltech101(data_root='data/Caltech', split='train')  # train dataset
        Dataset Caltech101
               Number of samples:  3060
               Number of categories:       102
               Root of dataset:    data/Caltech
```